### PR TITLE
fix(radio): fix the baseline of radio buttons

### LINKED ISSUE #357: Radio button and checkbox baselines don't align
When you put radio buttons and checkboxes side-by-side, they have different baselines.

cc: @jelbourn

### PR DESCRIPTION
**Pull request by hansl**
Fix the `ngModel` on the checkbox as well.

Fixes #357, #638.

### CODE REVIEW

**Comment by hansl** [comment]
R: @jelbourn @robertmesserle @kara

**Comment by hansl** [comment]
Also added a new demo page for baseline. Please use it :)

File: src/components/radio/radio.scss [hunk 1]
```diff
@@ -74,16 +75,26 @@ md-radio-button {
...
+.md-radio-label-content.md-radio-align-end {
+  order: -1;
+  padding-left: 0;
+  padding-right: $md-toggle-padding;
 
   [dir='rtl'] & {
-    float: left;
-    padding-right: $md-toggle-padding;
-    padding-left: 0;
+    order: 0;
```
Comment (robertmesserle) [inline src/components/radio/radio.scss]: Wouldn't this still need the padding styles?

Comment (hansl) [inline src/components/radio/radio.scss]: Fixed.

**Comment by robertmesserle** [comment]
LGTM

File: src/components/radio/radio.ts [hunk 2]
```diff
@@ -333,6 +335,17 @@ export class MdRadioButton implements OnInit {
     }
   }
 
+  private _align: 'start' | 'end';
+
+  @Input()
+  get align(): 'start' | 'end' {
+    return this._align || (this.radioGroup != null && this.radioGroup.align) || 'start';
```
Comment (jelbourn) [inline src/components/radio/radio.ts]: Unit test for getting `align` from the parent?

Comment (hansl) [inline src/components/radio/radio.ts]: Donezeo.

**Comment by jelbourn** [comment]
LGTM

**Pull request merged**